### DISCUSSION TITLE
An alternative style of active route

### DISCUSSION THREAD

**Discussion by Ciutnik** [devemux86/cruiser#421] [Q&A] [answered]
I have always wondered how to combine the view of the active route with the view of the street I am driving on on the map. I have known since Kurviger that the route image cannot be in the background - under the street drawing.

I currently use a wide, semi-transparent route, and this works reasonably well. But perhaps it would be better to have the route as two parallel strips, with a completely transparent strip in between (e.g. in a 1:2:1 ratio)? I could turn off the transparency and the street (mainly its category) would still be perfectly visible, which is something I miss a bit 🙂

**Comment by devemux86** [comment, marked as answer]
> I have known since Kurviger that the route image cannot be in the background - under the street drawing.

It cannot be under the streets (i.e. map) or it would be invisible, it is just under the street labels.

The lines with a border work a little differently, there are 2 lines: a thick line below, a thin line above.
If we make only the thin line transparent, then we will see the border below, not the street.
If both lines become transparent, it should be similar to the current appearance.

<img src="https://github.com/devemux86/cruiser/assets/3484020/533b7aa5-8ee3-4776-a895-ed9d270ff0a5" width="50%"/>

**Reply by Ciutnik** [reply]
Thanks for the clarification. I now recall that on some other occasion you explained that, contrary to appearances, double lines are not actually drawn side by side, but one on top of the other. Whereby the top one can overlap but not cut out the bottom one 🙂

All clear. Thanks again 👍

**Reply by devemux86** [reply]
> double lines are not actually drawn side by side, but one on top of the other. Whereby the top one can overlap but not cut out the bottom one

Correctly, this is how all map engines draw the road lines (and overlays).

Note: in the past we had such appearance for the route overlay line (i.e. with darker border),
but then we introduced the transparency in the route color, so we had to remove the border.

Nevertheless, it's an interesting exercise to solve in the future when there's free time from the other features. :slightly_smiling_face:

**Reply by Ciutnik** [reply]
> in the past we had such appearance for the route overlay line (i.e. with darker border), but then we introduced the transparency in the route color

Yeah! Now I remember! That's when we talked about it! It was years ago so it slipped my mind. So many things have happened since then... 🙂

**Reply by DolomitesTom** [reply]
> when there's free time from the other features. 🙂

Free time, with this long list? 😆

The speed information in the track details is already great. If now average speeds with and without psising is also included, it's awesome, unbeatable 😁.
